lib/protobuf/field/bytes_field.rb

```diff
@@ -50,33 +49,17 @@ def wire_type
         ::Protobuf::WireType::LENGTH_DELIMITED
       end
 
-      private
-
-      ##
-      # Private Instance Methods
-      #
-
-      def define_setter
-        field = self
-        method_name = field.setter
-
-        message_class.class_eval do
-          define_method(method_name) do |val|
-            @encode = nil
-            case val
-            when String, Symbol
-              @values[field.name] = "#{val}"
-            when NilClass
-              @values.delete(field.name)
-            when ::Protobuf::Message
-              @values[field.name] = val.dup
-            else
-              fail TypeError, "Unacceptable value #{val} for field #{field.name} of type #{field.type_class}"
-            end
-          end
+      def coerce!(value)
+        case value
+        when String, Symbol
```

Comment:
If you want to ensure you clone, then I would split it into two cases, one is `String` where you call `dup` and the other is `Symbol` where you call `to_s`. Better to reduce the # of method calls wherever we can.